src/Analyzer/Pass/Expression/LogicInversion.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * @author Medvedev Alexandr https://github.com/lexty <alexandr.mdr@gmail.com>
+ */
+namespace PHPSA\Analyzer\Pass\Expression;
+
+use PhpParser\Node\Expr\BooleanNot;
+use PHPSA\Analyzer\Pass\AnalyzerPassInterface;
+use PHPSA\Context;
+
+class LogicInversion implements AnalyzerPassInterface
+{
+    protected static $map = [
+        'Expr_BinaryOp_Equal' => ['!=', '=='],
+        'Expr_BinaryOp_NotEqual' => ['==', '!='],
+        'Expr_BinaryOp_Identical' => ['===', '!=='],
+        'Expr_BinaryOp_NotIdentical' => ['!==', '==='],
+        'Expr_BinaryOp_Greater' => ['>', '<='],
+        'Expr_BinaryOp_GreaterOrEqual' => ['>=', '<'],
+        'Expr_BinaryOp_Smaller' => ['<', '>='],
+        'Expr_BinaryOp_SmallerOrEqual' => ['<=', '>'],
+    ];
+
+    /**
+     * @param BooleanNot $expr
+     * @param Context $context
+     * @return bool
+     */
+    public function pass(BooleanNot $expr, Context $context)
+    {
+        if ($expr->expr->getType() === 'Expr_BooleanNot') {
```

Comment:
this is not needed we have another analyzer for it (called MultipleUnaryOperators)